File: insert_links.R
library(RMySQL)
library(jsonlite)

SECRETS_PATH = "/Users/roemerbakker/ownCloud/HvA/Data Processing And Storage/Individual Assignment/secrets.json"
secrets <- read_json(path = SECRETS_PATH)


DB_HOST <- secrets$db_host
DB_USER <- secrets$db_user
DB_PASSWD <- secrets$db_pass
DB_DATABASE <- secrets$db_database
DB_PORT <- secrets$db_port
DATA_PATH = "/Users/roemerbakker/ownCloud/HvA/Data Processing And Storage/Individual Assignment/data/links.csv"

data <- read.csv2(file=DATA_PATH, sep = ",")

dbCon <- dbConnect(
  MySQL(),
  user = DB_USER,
  password = DB_PASSWD,
  host = DB_HOST,
  port = DB_PORT,
  dbname = DB_DATABASE
)

for (row in rownames(data)) {
  if (is.na(data[row, "movieId"]))
  {
    next;
  }
  if (is.na(data[row, "tmdbId"]))
  {
    next;
  }
  if (is.na(data[row, "imdbId"]))
  {
    next;
  }
  
  dbSendQuery(dbCon, paste(
    "INSERT INTO `movielens_tmdb_imdb` (`id`, `tmdb_id`, `imdb_id`) VALUES (", data[row, "movieId"], ", ", data[row, "tmdbId"], ", ", data[row, "imdbId"], ");"
    )
  )
  print(paste("Inserted movie ", data[row, "movieId"]))
}

File: insert_links.py
import pymysql.cursors
import csv

DB_HOST = "127.0.0.1"
DB_USER = "homestead"
DB_PASS = "secret"
DB_PORT = 33060
DB_DATABASE = "hva_bigdata_indv_assignment"
def db_connect(host, user, passwd, db, port):
	connection = pymysql.connect(
			host=host,
			user=user,
			password=passwd,
			db=db,
			port=port,
			charset='utf8mb4',
			cursorclass=pymysql.cursors.DictCursor
		)

	return connection

reader = csv.reader(open('data/links.csv', 'rb'))
headers = reader.next()
db_connection = db_connect(DB_HOST, DB_USER, DB_PASS, DB_DATABASE, DB_PORT)

for line in reader:
	print "1: " + line[0]
	print "2: " + line[1]
	print "3: " + line[2]
	try:
		cursor = db_connection.cursor()
		print("Inserting row. TMDB ID: " + line[1])
		sql = "INSERT INTO `movielens_tmdb_imdb` (`id`, `tmdb_id`, `imdb_id`) VALUES (%d, %d, %d);"
		cursor.execute( sql, ( int( line[0] ), int( line[1] ), int( line[2] ) ) )
		cursor.commit()
	finally:
		print("Inserted ID: " + line[0])

db_connection.close()